controllers/spec/common.go

```diff
@@ -431,8 +448,61 @@ func MakeLivenessProbe(liveness *v1alpha1.Liveness) *corev1.Probe {
 	}
 }
 
-func getLegacyDownloadCommand(downloadPath, componentPackage string, authProvided, tlsProvided bool,
-	tlsConfig TLSConfig, authConfig *v1alpha1.AuthConfig) []string {
+func makeCleanUpJob(objectMeta *metav1.ObjectMeta, container *corev1.Container, volumes []corev1.Volume, labels map[string]string, policy v1alpha1.PodPolicy) *v1.Job {
+	temp := makePodTemplate(container, volumes, labels, policy, nil)
+	temp.Spec.RestartPolicy = corev1.RestartPolicyOnFailure
+	var ttlSecond int32
+	return &v1.Job{
+		ObjectMeta: *objectMeta,
+		Spec: v1.JobSpec{
+			Template:                *temp,
+			TTLSecondsAfterFinished: &ttlSecond,
+		},
+	}
+}
+
+func TriggerCleanup(ctx context.Context, k8sclient client.Client, restClient rest.Interface, config *rest.Config, job *v1.Job) error {
+	pods := &corev1.PodList{}
+	err := k8sclient.List(ctx, pods, client.InNamespace(job.Namespace), client.MatchingLabels{
+		"job-name": job.Name,
+	})
+	if err != nil || len(pods.Items) == 0 {
+		return errors.New("error list cleanup pod")
+	}
+	command := []string{"sh", "-c", "kill -s INT 1"}
+	res := restClient.Post().
+		Resource("pods").
+		Namespace(job.Namespace).
+		Name(pods.Items[0].Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Command:   command,
+			Container: CleanupContainerName,
+			Stdin:     false,
+			Stderr:    true,
+			Stdout:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+
+	fmt.Printf("query is %v", res.URL().Query())
```

Comment:
maybe change it to `logger` with the proper logging level?

Reply:
there are for debugging purpose, removed